src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -0,0 +1,546 @@
+                " not match <coefficient>*<operator>"
+            )
+        # parse the coefficient into complex
+        try:
+            coef = complex(str_coef.replace(" ", ""))
+        except ValueError:
+            raise ValueError(f"Could not parse the coefficient {str_coef}")
+
+        result_term = PauliTerm.identity() * coef
+        if str_op == "I":
+            assert isinstance(result_term, PauliTerm)
+            return result_term
+
+        for op in str_op.split("*"):
+            result_term *= PauliTerm(op)
+
+        assert isinstance(result_term, PauliTerm)
+        return result_term
+
+    @staticmethod
+    def identity() -> "PauliTerm":
+        return PauliTerm("I0", 1.0)
+
+    def copy(self, new_coefficient: CoefficientTypes = None) -> "PauliTerm":
+        """
+        Properly creates a new PauliTerm, with a completely new dictionary
+        of operators
+        """
+        new_term = PauliTerm.identity()  # create new object
+        # manually copy all attributes over
+        for key, val in self.__dict__.items():
+            if isinstance(val, (dict, list, set)):  # mutable types
+                new_term.__dict__[key] = copy.copy(val)
+            else:  # immutable types
+                new_term.__dict__[key] = val
+
+        if new_coefficient:
+            new_term.coefficient = new_coefficient
+
+        return new_term
+
+    @property
+    def qubits(self) -> Set[int]:
+        """
+        Returns the list of qubit indices associated with this term.
+        """
+        return set(self._ops.keys())
+
+    @property
+    def is_ising(self) -> bool:
+        """
+        Returns whether the term represents an ising model
+        (i.e. contains only Z terms)
+        """
+
+        return set(self._ops.values()) == {"Z"}
+
+    @property
+    def circuit(self) -> Circuit:
+        """
+        Returns the circuit implementing this Pauli Term. Since the public API
+        treats the object as immutable, we store the circuit representation when
+        the function is called for the first time, for efficiency.
+        """
+        if not hasattr(self, "_circuit"):
+            self._circuit = Circuit(
+                [cast(Operation, builtin_gate_by_name(op)(index)) for op, index in self]
+            )
+
+        return self._circuit
+
+    def operations_as_set(self) -> FrozenSet[Tuple[int, str]]:
+        return frozenset(self._ops.items())
+
+    def __len__(self) -> int:
+        return len(self._ops)
+
+    def __getitem__(self, i: int) -> str:
+        if i not in self._ops.keys():
+            warnings.warn("Index not in term! Identity is returned.", UserWarning)
+
+        return self._ops.get(i, "I")
+
+    def __iter__(self) -> Iterator[Tuple[str, int]]:
+        for i in self.qubits:
+            yield self[i], i
+
+    def __hash__(self) -> int:
+        assert isinstance(self.coefficient, complex)
+        return hash(
```

Comment:
I simply carried over how it was implemented here: https://github.com/rigetti/pyquil/blob/master/pyquil/paulis.py#L116
Since I am not well-versed in precision, I simply trusted the developers. :sweat_smile: